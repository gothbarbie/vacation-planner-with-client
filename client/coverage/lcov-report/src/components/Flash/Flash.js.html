<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Flash/Flash.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Flash</a> Flash.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/6</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
&nbsp;
import React, { Component } from 'react'
import { connect } from 'react-redux'
import classnames from 'classnames'
import type { MapStateToProps } from 'react-redux'
&nbsp;
import * as actions from './flashActions'
&nbsp;
import './Flash.css'
&nbsp;
type Props = {
  flash: {
    type: void | string,
    message: void | string,
  },
  closeFlash: Function,
}
&nbsp;
export class Flash extends Component&lt;Props&gt; {
  renderMessage () {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (this.props.flash) {
<span class="cstat-no" title="statement not covered" >      return (</span>
        &lt;section
          className={classnames('flash', {
            'flash-success': this.props.flash.type === 'success',
            'flash-danger': this.props.flash.type === 'danger',
            'flash-warning': this.props.flash.type === 'warning',
          })}
          onClick={this.props.closeFlash}&gt;
          {this.props.flash.message}
        &lt;/section&gt;
      )
    }
    return null
  }
&nbsp;
  render () {
    return this.renderMessage()
  }
}
&nbsp;
const mapStateToProps: MapStateToProps&lt;*, *, *&gt; = <span class="fstat-no" title="function not covered" >({</span> flash }) =&gt; {
<span class="cstat-no" title="statement not covered" >  return {</span>
    flash,
  }
}
&nbsp;
export default connect(mapStateToProps, actions)(Flash)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Mar 11 2018 12:53:11 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
