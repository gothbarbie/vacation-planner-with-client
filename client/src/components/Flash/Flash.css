.flash {
  display: flex;
  justify-content: center;
  align-items: center;
  background: #f2f2f2;
  padding: .5rem;
  border: 2px solid #bdbdbd;
  font-family: 'Avenir Next', sans-serif; }

.flash-success {
  background: #6fcf97;
  border: 2px solid #27ae60; }

.flash-danger {
  background: #eb5757;
  border: 2px solid #993939; }

.flash-warning {
  background: #f2c94c;
  border: 2px solid #f2994a; }
