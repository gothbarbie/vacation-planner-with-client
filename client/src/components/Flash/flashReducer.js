import { FLASH_MESSAGE, FLASH_RESET } from '../../actions/types'

export default function (state = null, action) {
  switch (action.type) {
    case FLASH_MESSAGE:
      return action.payload

    case FLASH_RESET:
      return null

    default:
      return state
  }
}
