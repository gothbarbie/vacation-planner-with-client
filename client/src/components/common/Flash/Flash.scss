@import '../../_colors';

.flash {
  display: flex;
  justify-content: center;
  align-items: center;
  background: $color-concrete;
  padding: .5rem;
  border: 2px solid $color-silver;
  font-family: 'Avenir Next', sans-serif;
}

.flash-success {
  background: $color-de-york;
  border: 2px solid $color-jungle-green;
}

.flash-danger {
  background: $color-burnt-sienna;
  border: 2px solid $color-sanguine-brown;
}

.flash-warning {
  background: $color-cream-can;
  border: 2px solid $color-jaffa;
}
