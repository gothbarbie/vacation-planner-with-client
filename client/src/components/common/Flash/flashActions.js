import { FLASH_RESET } from '../../../actions/types'

export const closeFlash = () => dispatch => {
  dispatch({
    type: FLASH_RESET,
  })
}
