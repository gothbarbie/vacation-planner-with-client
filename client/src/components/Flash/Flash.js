// @flow

import React, { Component } from 'react'
import { connect } from 'react-redux'
import classnames from 'classnames'

import * as actions from './flashActions'

import './Flash.css'

type Props = {
  flash: {
    type: void | string,
    message: void | string,
  },
  closeFlash: Function,
}

export class Flash extends Component<Props> {
  renderMessage () {
    if (this.props.flash) {
      return (
        <section
          className={classnames('flash', {
            'flash-success': this.props.flash.type === 'success',
            'flash-danger': this.props.flash.type === 'danger',
            'flash-warning': this.props.flash.type === 'warning',
          })}
          onClick={this.props.closeFlash}>
          {this.props.flash.message}
        </section>
      )
    }
    return null
  }

  render () {
    return this.renderMessage()
  }
}

export const mapStateToProps = ({ flash }: Object) => {
  return {
    flash,
  }
}

export default connect(mapStateToProps, actions)(Flash)
