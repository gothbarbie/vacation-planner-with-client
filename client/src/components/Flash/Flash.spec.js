import React from 'react'
import { shallow } from 'enzyme'

import { Flash } from './Flash'
import {Â mapStateToProps } from './Flash'

describe('Flash', () => {
  const closeFlash = jest.fn()

  const props = {
    flash: {
      type: '',
      message: 'Default',
    },
    closeFlash,
  }
  const flash = shallow(<Flash {...props} />)

  it('renders (default)', () => {
    expect(flash).toMatchSnapshot()
  })

  it('renders (success)', () => {
    flash.setProps({
      flash: {
        type: 'success',
        message: 'Success!',
      },
    })
    expect(flash).toMatchSnapshot()
  })

  it('renders (danger)', () => {
    flash.setProps({
      flash: {
        type: 'danger',
        message: 'Danger!',
      },
    })
    expect(flash).toMatchSnapshot()
  })

  it('renders (warning)', () => {
    flash.setProps({
      flash: {
        type: 'warning',
        message: 'Warning!',
      },
    })
    expect(flash).toMatchSnapshot()
  })

  describe('#closeFlash', () => {
    it('calls closeFlash() on click', () => {
      flash.find('.flash').simulate('click')
      expect(closeFlash).toHaveBeenCalled()
    })

    it('stops rendering flash-component', () => {
      flash.setProps({
        flash: null,
      })
      expect(flash).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('returns flash as flash', () => {
      const flash = { type: 'success', message: 'Success!' }
      expect(mapStateToProps({ flash })).toEqual({ flash })
    })
  })
})
