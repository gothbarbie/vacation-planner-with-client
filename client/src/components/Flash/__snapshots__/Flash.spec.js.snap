// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flash #closeFlash stops rendering flash-component 1`] = `""`;

exports[`Flash renders (danger) 1`] = `
<section
  className="flash flash-danger"
  onClick={[Function]}
>
  Danger!
</section>
`;

exports[`Flash renders (default) 1`] = `
<section
  className="flash"
  onClick={[Function]}
>
  Default
</section>
`;

exports[`Flash renders (success) 1`] = `
<section
  className="flash flash-success"
  onClick={[Function]}
>
  Success!
</section>
`;

exports[`Flash renders (warning) 1`] = `
<section
  className="flash flash-warning"
  onClick={[Function]}
>
  Warning!
</section>
`;
